optimum/habana/diffusers/pipelines/qwenimage/pipeline_qwenimage.py

```diff
@@ -0,0 +1,442 @@
+# Copyright 2025 Qwen-Image Team and The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import inspect
+from typing import Any, Callable, Dict, List, Optional, Union
+import types
+import numpy as np
+import torch
+from transformers import (
+    Qwen2_5_VLForConditionalGeneration,
+    Qwen2Tokenizer,
+)
+
+from diffusers.models import AutoencoderKLQwenImage, QwenImageTransformer2DModel
+from diffusers.schedulers import FlowMatchEulerDiscreteScheduler
+from diffusers.utils import (
+    is_torch_xla_available,
+    logging,
+    replace_example_docstring,
+)
+
+from ..pipeline_utils import GaudiDiffusionPipeline
+from diffusers.pipelines.qwenimage.pipeline_output import QwenImagePipelineOutput
+from diffusers.pipelines.qwenimage.pipeline_qwenimage import QwenImagePipeline,calculate_shift,retrieve_timesteps
+from ....transformers.gaudi_configuration import GaudiConfig
+from ...models.attention_processor import GaudiQwenDoubleStreamAttnProcessor2_0
+from ...models.qwenimage_transformer import QwenImageTransformer2DModelGaudi
+from ....utils import HabanaProfile
+
+logger = logging.get_logger(__name__)  # pylint: disable=invalid-name
+
+EXAMPLE_DOC_STRING = """
+    Examples:
+        ```python
+        >>> import torch
+        >>> from optimum.habana.diffusers import GaudiQwenImagePipeline
+
+        >>> pipe = GaudiQwenImagePipeline.from_pretrained(       
+        ...    "Qwen/Qwen-Image",
+        ...     torch_dtype=torch.bfloat16,
+        ...     use_habana=True,
+        ...     use_hpu_graphs=True,
+        ...     gaudi_config="Habana/stable-diffusion",
+        ... )
+
+        >>> prompt = "A cat holding a sign that says hello world"
+        >>> # Depending on the variant being used, the pipeline call will slightly vary.
+        >>> # Refer to the pipeline documentation for more details.
+        >>> image = pipe(prompt, num_inference_steps=50).images[0]
+        >>> image.save("qwenimage.png")
+        ```
+"""
+
+class GaudiQwenImagePipeline(GaudiDiffusionPipeline, QwenImagePipeline):
+    r"""
+    Adapted from: https://github.com/huggingface/diffusers/blob/main/src/diffusers/pipelines/qwenimage/pipeline_qwenimage.py#L132
```

Comment:
It is better to use a specific commit or tag for the code link you refer to, as "main" always points to the latest and gets updated frequently. for example use link :https://github.com/huggingface/diffusers/blob/commit_id_xxx/src/diffusers/pipelines/qwenimage/pipeline_qwenimage.py#L132